benchmark/evaluation/compare_near_eq_and_exact.py

```diff
@@ -0,0 +1,292 @@
+    for i in range(n_inner_samples):
+        Ws[i] = inner_sample(noneq_sim, x, v, n_steps, marginal)
+    return Ws
+
+
+def collect_inner_samples_until_threshold(x, v, noneq_sim, marginal="full", initial_size=100, batch_size=5,
+                                          n_steps=1000, threshold=0.1, max_samples=1000):
+    """Collect up to max_samples trajectories, potentially fewer if stdev of estimated log(rho(x,v) / pi(x,v)) is below threshold."""
+    Ws = []
+
+    # collect initial samples
+    for _ in range(initial_size):
+        Ws.append(inner_sample(noneq_sim, x, v, n_steps, marginal))
+
+    # keep adding batches until either stdev threshold is reached or budget is reached
+    while (stdev_log_rho_pi(np.array(Ws)) > threshold) and (len(Ws) <= (max_samples - batch_size)):
+        for _ in range(batch_size):
+            Ws.append(inner_sample(noneq_sim, x, v, n_steps, marginal))
+
+    # warn user if stdev threshold was not met
+    if (stdev_log_rho_pi(np.array(Ws)) > threshold):
+        message = "stdev_log_rho_pi(Ws) > threshold\n({:.3f} > {:.3f})".format(stdev_log_rho_pi(np.array(Ws)), threshold)
+        warnings.warn(message, RuntimeWarning)
+
+
+    return np.array(Ws)
+
+
+def sample_from_rho(noneq_sim, n_steps=1000):
+    # (x0, v0) drawn from pi
+    x0 = noneq_sim.sample_x_from_equilibrium()
+    v0 = noneq_sim.sample_v_given_x(x0)
+
+    # (x, v) drawn from rho
+    W_shad_forward = noneq_sim.accumulate_shadow_work(x0, v0, n_steps)["W_shad"]
+    x = get_state_as_mdtraj(noneq_sim.simulation)
+    v = noneq_sim.simulation.context.getState(getVelocities=True).getVelocities(asNumpy=True)
+
+    return {'x': x,
+            'v': v,
+            'W_shad_forward': W_shad_forward,
+            }
+
+
+def outer_sample_naive(index=0, noneq_sim=None, marginal="full", n_inner_samples=100, n_steps=1000):
+    rho_sample = sample_from_rho(noneq_sim, n_steps)
+    x, v, W_shad_forward = rho_sample['x'], rho_sample['v'], rho_sample['W_shad_forward']
+
+    Ws = collect_inner_samples_naive(x, v, noneq_sim, marginal, n_steps, n_inner_samples)
+
+    return {
+        "xv": (x, v),
+        "Ws": Ws,
+        "estimate": estimate_from_work_samples(Ws),
+        "W_shad_forward": W_shad_forward
+    }
+
+
+def outer_sample_adaptive(noneq_sim=None, marginal="full", n_steps=1000, initial_size=100, batch_size=5, threshold=0.1,
+                          max_samples=1000):
+    rho_sample = sample_from_rho(noneq_sim, n_steps)
+    x, v, W_shad_forward = rho_sample['x'], rho_sample['v'], rho_sample['W_shad_forward']
+
+    Ws = collect_inner_samples_until_threshold(x, v, noneq_sim, marginal, initial_size, batch_size, n_steps, threshold,
+                                               max_samples)
+
+    return {
+        "xv": (x, v),
+        "Ws": Ws,
+        "estimate": estimate_from_work_samples(Ws),
+        "W_shad_forward": W_shad_forward
+    }
+
+
+def noneq_sim_factory(testsystem_name, scheme, dt, collision_rate):
```

Comment:
In cases where some arguments are bearing units and some are unitless (why would you do that???) it's CRITICAL to document your parameters and check that the user hasn't mis-specified any that should or should not bear units. It's important to remember that OpenMM doesn't check for unit compatibility, so it's easy to make mistakes like passing `dt = 1  * unit.femtosecond` and then this becomes `1 * unit.femtosecond**2` and would be stripped of units to be much smaller than you intended.

Reply:
Addressed in https://github.com/choderalab/integrator-benchmark/pull/45/commits/3145e26ff0803b4fd3632866d0d8c147c3e41c34 (The unitless dt issue was a hack around a bug i had pickling / unpickling the "experiment" tuple, but it was unnecessary.)